cpp/src/parquet/properties.h

```diff
@@ -135,14 +136,13 @@ static constexpr int64_t DEFAULT_WRITE_BATCH_SIZE = 1024;
 static constexpr int64_t DEFAULT_MAX_ROW_GROUP_LENGTH = 1024 * 1024;
 static constexpr bool DEFAULT_ARE_STATISTICS_ENABLED = true;
 static constexpr int64_t DEFAULT_MAX_STATISTICS_SIZE = 4096;
-static constexpr Encoding::type DEFAULT_ENCODING = Encoding::PLAIN;
 static const char DEFAULT_CREATED_BY[] = CREATED_BY_VERSION;
 static constexpr Compression::type DEFAULT_COMPRESSION_TYPE = Compression::UNCOMPRESSED;
 static constexpr bool DEFAULT_IS_PAGE_INDEX_ENABLED = false;
 
 class PARQUET_EXPORT ColumnProperties {
  public:
-  ColumnProperties(Encoding::type encoding = DEFAULT_ENCODING,
+  ColumnProperties(std::optional<Encoding::type> encoding = std::nullopt,
```

Comment:
Personally I'm ok with this, but should we make constructor compatible?

Reply:
Making it compatible requires us to write a default encoding value. We have to use UNKNOWN or UNDEFINED instead of PLAIN now. This could be dirty. Usually this constructor is used internally without any parameters supplied. So I think users will not be affected.

Reply:
Okay, this looks ok to me

Reply:
Hmm... if we want to switch to `std::optional` in `ColumnProperties` we should probably do so more consistently, instead of breaking compatibility for this single property. Can this be deferred to another issue and PR?

Reply:
> Hmm... if we want to switch to `std::optional` in `ColumnProperties` we should probably do so more consistently, instead of breaking compatibility for this single property. Can this be deferred to another issue and PR?

Do you mean we can use `Encoding::UNKNOWN` as the default to fix the current issue without breaking the compatibility?

Reply:
For this PR, yes.